index.js

```diff
@@ -316,6 +316,10 @@ function HttpRequest(options) {
 			}
 		};
 	}
+
+	if ('onerror' in xhr && !xhr.onreadystatechange) {
+		xhr.onerror = onLoad;
```

Comment:
Why don't we use `onreadystatechange` in every case in the first place?
That's the only thing which is supported by every browser (except maybe IE).